var/spack/repos/builtin/packages/maqao/package.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from llnl.util.filesystem import find_system_libraries
+
+from spack.package import *
+from spack.util.executable import *
+
+
+class Maqao(CMakePackage):
+    """MAQAO (Modular Assembly Quality Analyzer and Optimizer) is a performance
+    analysis and optimization framework operating at binary level with a focus
+    on core performance. Its main goal is to guide application developers
+    along the optimization process through synthetic reports and hints."""
+
+    homepage = "https://maqao.exascale-computing.eu/"
+    git = "https://gitlab.exascale-computing.eu/MAQAO/MAQAO.git"
+    url = "https://gitlab.exascale-computing.eu/MAQAO/MAQAO/repository/archive.tar.bz2?ref=master"
+
+    maintainers = ["cvalensi", "yaspr"]
+    executables = ["maqao"]
+
+    version("master", branch="master")
+
+    depends_on("cmake@2.8.12:", type="build")
+    depends_on('gcc@4.8.1: languages="c,c++"', when="%gcc")
+
+    # From 'profiles' subdirectory
+    variant(
+        "profile",
+        multi=False,
+        default="default",
+        values=("default", "release", "release.intel64-xeonphi"),
+        description="What profile to build",
+    )
+
+    # From 'src/arch.h' source file
+    variant(
+        "arch",
+        multi=True,
+        default="x86_64",
+        values=("ia32", "x86_64", "k1om", "arm", "arm64", "power", "thumb"),
+        description="What architectures to build",
+    )
+
+    variant(
+        "exclude_uarch",
+        multi=True,
+        default="none",
+        description="Microarchitectures to exclude from build",
+    )
+
+    variant("strip", default=False, description="Strip the MAQAO binary")
+
+    variant(
+        "lua",
+        multi=False,
+        default="luajit",
+        values=("lua", "luajit"),
+        description="Lua compiler to use",
+    )
+    depends_on("lua", type=("build", "run"), when="lua=lua")
+    depends_on("lua-luajit", type=("build", "run"), when="lua=luajit")
+
+    variant("xlsx", default=False, description="Enable .xlsx output for ONE-View")
```

Comment:
Can we avoid intertwining variants and other directives, if kept at the same level and not nested in a `with when`?

Reply:
Hmm, is`lua-luajit` (LuaJIT) independent from `lua` (LuaC)?

Reply:
What I mean is: put all `variant` directives together, and then all `depends_on`. In that way reading a `package.py` is easier. The only exception I have seen in packages to this way of writing directives is grouping under a `with when` context manager.

Reply:
Got it, thanks.